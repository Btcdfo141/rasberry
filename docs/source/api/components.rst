:mod:`spencerassistant.components`
===============================

air\_quality
--------------------------------------------

.. automodule:: spencerassistant.components.air_quality
    :members:
    :undoc-members:
    :show-inheritance:

alarm\_control\_panel
--------------------------------------------

.. automodule:: spencerassistant.components.alarm_control_panel
    :members:
    :undoc-members:
    :show-inheritance:

binary\_sensor
--------------------------------------------

.. automodule:: spencerassistant.components.binary_sensor
    :members:
    :undoc-members:
    :show-inheritance:

camera
---------------------------

.. automodule:: spencerassistant.components.camera
    :members:
    :undoc-members:
    :show-inheritance:

calendar
---------------------------

.. automodule:: spencerassistant.components.calendar
    :members:
    :undoc-members:
    :show-inheritance:

climate
---------------------------

.. automodule:: spencerassistant.components.climate
    :members:
    :undoc-members:
    :show-inheritance:

conversation
---------------------------

.. automodule:: spencerassistant.components.conversation
    :members:
    :undoc-members:
    :show-inheritance:

cover
---------------------------

.. automodule:: spencerassistant.components.cover
    :members:
    :undoc-members:
    :show-inheritance:

device\_tracker
---------------------------

.. automodule:: spencerassistant.components.device_tracker
    :members:
    :undoc-members:
    :show-inheritance:

fan
---------------------------

.. automodule:: spencerassistant.components.fan
    :members:
    :undoc-members:
    :show-inheritance:

light
---------------------------

.. automodule:: spencerassistant.components.light
    :members:
    :undoc-members:
    :show-inheritance:

lock
---------------------------

.. automodule:: spencerassistant.components.lock
    :members:
    :undoc-members:
    :show-inheritance:

media\_player
---------------------------

.. automodule:: spencerassistant.components.media_player
    :members:
    :undoc-members:
    :show-inheritance:

notify
---------------------------

.. automodule:: spencerassistant.components.notify
    :members:
    :undoc-members:
    :show-inheritance:

remote
---------------------------

.. automodule:: spencerassistant.components.remote
    :members:
    :undoc-members:
    :show-inheritance:

switch
---------------------------

.. automodule:: spencerassistant.components.switch
    :members:
    :undoc-members:
    :show-inheritance:

sensor
-------------------------------------

.. automodule:: spencerassistant.components.sensor
    :members:
    :undoc-members:
    :show-inheritance:

vacuum
-------------------------------------

.. automodule:: spencerassistant.components.vacuum
    :members:
    :undoc-members:
    :show-inheritance:

water\_heater
-------------------------------------

.. automodule:: spencerassistant.components.water_heater
    :members:
    :undoc-members:
    :show-inheritance:

weather
---------------------------

.. automodule:: spencerassistant.components.weather
    :members:
    :undoc-members:
    :show-inheritance:

webhook
---------------------------

.. automodule:: spencerassistant.components.webhook
    :members:
    :undoc-members:
    :show-inheritance:
