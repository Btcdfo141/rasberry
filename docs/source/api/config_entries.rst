.. _config_entries_module:

:mod:`spencerassistant.config_entries`
-----------------------------------

.. automodule:: spencerassistant.config_entries
    :members:
