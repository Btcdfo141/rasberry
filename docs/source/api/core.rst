.. _core_module:

:mod:`spencerassistant.core`
-------------------------

.. automodule:: spencerassistant.core
    :members: