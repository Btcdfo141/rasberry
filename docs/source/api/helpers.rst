:mod:`spencerassistant.helpers`
============================

.. automodule:: spencerassistant.helpers
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.aiohttp\_client
-------------------------------------

.. automodule:: spencerassistant.helpers.aiohttp_client
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.area\_registry
------------------------------------

.. automodule:: spencerassistant.helpers.area_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.check\_config
-----------------------------------

.. automodule:: spencerassistant.helpers.check_config
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.collection
--------------------------------

.. automodule:: spencerassistant.helpers.collection
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.condition
-------------------------------

.. automodule:: spencerassistant.helpers.condition
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_flow
-----------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_entry\_oauth2\_flow
-------------------------------------------------

.. automodule:: spencerassistant.helpers.config_entry_oauth2_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.config\_validation
----------------------------------------

.. automodule:: spencerassistant.helpers.config_validation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.data\_entry\_flow
---------------------------------------

.. automodule:: spencerassistant.helpers.data_entry_flow
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.debounce
------------------------------

.. automodule:: spencerassistant.helpers.debounce
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.deprecation
---------------------------------

.. automodule:: spencerassistant.helpers.deprecation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.device\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.device_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.discovery
-------------------------------

.. automodule:: spencerassistant.helpers.discovery
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.dispatcher
--------------------------------

.. automodule:: spencerassistant.helpers.dispatcher
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity
----------------------------

.. automodule:: spencerassistant.helpers.entity
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_component
---------------------------------------

.. automodule:: spencerassistant.helpers.entity_component
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_platform
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_registry
--------------------------------------

.. automodule:: spencerassistant.helpers.entity_registry
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entity\_values
------------------------------------

.. automodule:: spencerassistant.helpers.entity_values
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.entityfilter
----------------------------------

.. automodule:: spencerassistant.helpers.entityfilter
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.event
---------------------------

.. automodule:: spencerassistant.helpers.event
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.icon
--------------------------

.. automodule:: spencerassistant.helpers.icon
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.integration\_platform
-------------------------------------------

.. automodule:: spencerassistant.helpers.integration_platform
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.intent
----------------------------

.. automodule:: spencerassistant.helpers.intent
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.json
--------------------------

.. automodule:: spencerassistant.helpers.json
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.location
------------------------------

.. automodule:: spencerassistant.helpers.location
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.logging
-----------------------------

.. automodule:: spencerassistant.helpers.logging
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.network
-----------------------------

.. automodule:: spencerassistant.helpers.network
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.restore\_state
------------------------------------

.. automodule:: spencerassistant.helpers.restore_state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.script
----------------------------

.. automodule:: spencerassistant.helpers.script
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.service
-----------------------------

.. automodule:: spencerassistant.helpers.service
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.signal
-----------------------------

.. automodule:: spencerassistant.helpers.signal
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.state
---------------------------

.. automodule:: spencerassistant.helpers.state
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.storage
-----------------------------

.. automodule:: spencerassistant.helpers.storage
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.sun
-------------------------

.. automodule:: spencerassistant.helpers.sun
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.system\_info
----------------------------------

.. automodule:: spencerassistant.helpers.system_info
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.temperature
---------------------------------

.. automodule:: spencerassistant.helpers.temperature
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.template
------------------------------

.. automodule:: spencerassistant.helpers.template
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.translation
---------------------------------

.. automodule:: spencerassistant.helpers.translation
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.typing
----------------------------

.. automodule:: spencerassistant.helpers.typing
   :members:
   :undoc-members:
   :show-inheritance:

spencerassistant.helpers.update\_coordinator
-----------------------------------------

.. automodule:: spencerassistant.helpers.update_coordinator
   :members:
   :undoc-members:
   :show-inheritance:
