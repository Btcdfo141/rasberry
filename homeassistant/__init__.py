"""Init file for Home Assistant."""


test2
