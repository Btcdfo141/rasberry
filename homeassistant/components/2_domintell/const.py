"""Constants for the Domintell Integration integration."""

DOMAIN = "2_domintell"
