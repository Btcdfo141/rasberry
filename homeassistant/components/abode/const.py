"""Constants for the Abode Security System component."""
import logging

LOGGER = logging.getLogger(__package__)

DOMAIN = "abode"
ATTRIBUTION = "Data provided by goabode.com"

DEFAULT_CACHEDB = "jaraco_abode_cache.pickle"
CONF_POLLING = "polling"
