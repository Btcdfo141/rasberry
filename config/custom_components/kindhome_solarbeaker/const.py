from datetime import timedelta

from homeassistant.const import Platform

DOMAIN="kindhome_solarbeaker"
PLATFORMS = [Platform.COVER]

UPDATE_INTERVAL = timedelta(seconds=20)

SERVICE_UUID = "75c276c3-8f97-20bc-a143-b354244886d4"
TITLE="Kindhome Solarbeaker"

DATA_DEVICE = "device"
DATA_COOR = "coordinator"
