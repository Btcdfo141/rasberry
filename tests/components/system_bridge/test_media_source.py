"""Test the System Bridge integration."""

from unittest.mock import MagicMock

from syrupy.assertion import SnapshotAssertion
from syrupy.filters import paths

from homeassistant.components.media_source import URI_SCHEME, async_browse_media
from homeassistant.components.system_bridge.const import DOMAIN
from homeassistant.config_entries import ConfigEntryState
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr

from . import FIXTURE_UUID, setup_integration

from tests.common import MockConfigEntry


async def test_async_browse_media(
    hass: HomeAssistant,
    device_registry: dr.DeviceRegistry,
    snapshot: SnapshotAssertion,
    mock_config_entry: MockConfigEntry,
    mock_setup_entry: MagicMock,
) -> None:
    """Test async_browse_media."""
    await setup_integration(hass, mock_config_entry)

    assert mock_config_entry.state == ConfigEntryState.LOADED

    # Get device from device registry
    device = device_registry.async_get_or_create(
        config_entry_id=mock_config_entry.entry_id,
        identifiers={(DOMAIN, FIXTURE_UUID)},
    )
    assert device is not None

    browse_media_root = await async_browse_media(
        hass,
        f"{URI_SCHEME}{DOMAIN}",
    )

    assert browse_media_root.as_dict() == snapshot(
        name=f"{DOMAIN}_browse_media_root",
        exclude=paths("children", "media_content_id"),
    )

    # browse_media_entry = await async_browse_media(
    #     hass,
    #     f"{URI_SCHEME}{DOMAIN}~~{entry.entry_id}",
    # )

    # assert browse_media_entry.as_dict() == snapshot(name=f"{DOMAIN}_browse_media_entry")
