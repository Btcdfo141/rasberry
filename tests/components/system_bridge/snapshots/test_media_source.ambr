# serializer version: 1
# name: test_async_browse_media[system_bridge_browse_media]
  <BrowseMedia System Bridge (directory)>
# ---
# name: test_async_browse_media[system_bridge_browse_media_root]
  dict({
    'can_expand': True,
    'can_play': False,
    'children_media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_content_type': '',
    'not_shown': 0,
    'thumbnail': None,
    'title': 'System Bridge',
  })
# ---
# name: test_directory[system_bridge_media_source_directory]
  dict({
    'can_expand': True,
    'can_play': False,
    'children_media_class': None,
    'media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_content_type': '',
    'not_shown': 0,
    'thumbnail': None,
    'title': 'TestSystem - testdirectory',
  })
# ---
# name: test_entry[system_bridge_media_source_entry]
  dict({
    'can_expand': True,
    'can_play': False,
    'children_media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_content_type': '',
    'not_shown': 0,
    'thumbnail': None,
    'title': 'TestSystem',
  })
# ---
# name: test_file[system_bridge_media_source_file]
  dict({
    'mime_type': 'text/plain',
    'url': 'http://127.0.0.1:9170/api/media/file/data?token=abc-123-def-456-ghi&base=testdirectory&path=testfile.txt',
  })
# ---
# name: test_root[system_bridge_media_source_root]
  dict({
    'can_expand': True,
    'can_play': False,
    'children_media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_class': <MediaClass.DIRECTORY: 'directory'>,
    'media_content_type': '',
    'not_shown': 0,
    'thumbnail': None,
    'title': 'System Bridge',
  })
# ---
